### PR TITLE
Acceptance tests working

### PR DESCRIPTION
**Pull request by SarahJohnsonONS**
### What

`environment.py` - sets up a docker container to run the Flask app required to route requests sent as part of acceptance test scenarios. Container is stopped and removed  once all the tests have finished running. Dockerfile and `app.py`  in `features/docker/fake_backend`.

`temporary.feature` - contains three acceptance tests with different inputs.

`temporary.py` - step definitions required to run the tests in `temporary.feature`, and some utility functions for parsing docker logs.

### How to review

Sanity check, make sure you can run the acceptence tests by running `behave` in the repo root directory. I've added a `logs.txt` file showing the content of the docker logs for the three scenarios - suggestions for making these more useful welcome.

### Who can review

Anyone.

### CODE REVIEW

File: features/environment.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    context.session.headers = {}
+
+
+def after_all(context):
+    """
+    Stop and remove the docker container after all tests have finished running.
+    """
+    context.backend_container.stop()
+    context.backend_container.remove()
+    ...
```
Comment (mikeAdamss) [inline features/environment.py]: remove please

File: features/steps/temporary.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        f"http://127.0.0.1:5001{upload_service_endpoint_url}",
+        headers=context.request_headers,
+        json=context.json,
+    )
+    context.receiving_service_log = get_request_logs(
+        context.backend_container, context.request_id
+    )
+
+
+@Then('the backend receives a request to "{upload_service_endpoint}"')
```
Comment (mikeAdamss) [inline features/steps/temporary.py]: can you make it `service_endpoint`, there's an api to come and we can reuse this if we make it nice and generic.

File: logs.txt
Comment (mikeAdamss) [inline logs.txt:1]: I dont think we need this file

File: pyproject.toml [hunk 3]
```diff
@@ -9,6 +9,9 @@ include = ["dpypelines/schemas/**/*.json"]
 [tool.poetry.dependencies]
 python = ">=3.9.0,<3.10.0"
 dpytools = { git = "https://github.com/ONSdigital/dp-python-tools.git", branch="develop" }
+behave = "^1.2.6"
```
Comment (mikeAdamss) [inline pyproject.toml]: these are dev dependencies, you can move them down to `tool.poetry.group.dev.dependencies` also flask is only inside the docker container so we shouldnt need a package for it.

**Review by mikeAdamss — COMMENTED** [review]
minor comments, but can you also update `make install` to install the poetry dev dependencies then add a `make feature` command that runs it before calling behave.

File: features/environment.py (hunk 1 above)
Comment (SarahJohnsonONS) [reply features/environment.py]: If I remove the `after_all` function, run `behave`, then run it again, I get an error: `"Conflict. The container name "/fake_backend" is already in use by container "593c37d0ee6e1eec302455e549b7d13075ab8ef0c318eb8ed9772129dcf933c7". You have to remove (or rename) that container to be able to reuse that name."`

Comment (SarahJohnsonONS) [reply features/environment.py]: Oh, you meant the ellipsis XD

**Comment by SarahJohnsonONS** [comment]
> minor comments, but can you also update `make install` to install the poetry dev dependencies then add a `make feature` command that runs it before calling behave.

`make install` currently runs `poetry install`, which I thought installed all dependencies, including dev ones?

**Review by mikeAdamss — APPROVED** [review]
lgtm 👍

**Pull request merged**